components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -116,7 +127,16 @@ internal class MediaSessionServiceDelegate(
         }
 
         updateMediaSession(state)
-        updateNotification(state)
+        CoroutineScope(notificationDispatcher).launch() {
+            updateNotification(state)
+        }
+    }
+
+    private fun getNotificationDispatcher(): CoroutineDispatcher {
+        val artworkThread = HandlerThread("notificationThread").also {
+            it.start()
+        }
+        return Handler(artworkThread.looper).asCoroutineDispatcher("MediaSessionServiceDelegateDispatcher")
```

Comment:
I am not really sure what you are trying to achieve here. Could we run the loading of the artworks on the already existing dispatchers for IO? (`Dispatchers.IO`) and I assume the notification manager APIs are expected to be called on the main thread?

I wonder if something like this could work:
* We create a `MainScope` in the service - bound to the lifetime of the service. This will launch everything on the main thread.
* The `await()` call for the artwork already suspends and doesn't block the thread, so I think it's not needed to move this to any other thread explicitly.

Reply:
I'll double check this.  I think this was a constraint with retrieving the bitmap.  From my testing I had an issue with just using `Dispatcher.IO`.  I'll update.

Reply:
Yes, confirmed.  A handler have to be provided.  If I just use `Dispatchers.IO` this exception is thrown

Reply:
I'll create `MainScope` like you suggested.  Thanks

Reply:
Ah, yeah, right, luckily the `MainScope` will have a handler :)